src/components/EditOrgsForm.vue

```diff
@@ -0,0 +1,207 @@
+        <PvInputText v-else v-model="localOrgData.name" />
+      </div>
+      <div class="form-field w-full">
+        <label :class="{ 'font-light uppercase text-sm': !editMode }">Abbreviation</label>
+        <div v-if="!editMode" :class="{ 'text-xl': !editMode }">{{ serverOrgData.abbreviation ?? 'None' }}</div>
+        <PvInputText v-else v-model="localOrgData.abbreviation" />
+      </div>
+    </div>
+    <div class="form-field">
+      <label :class="{ 'font-light uppercase text-sm': !editMode }">Address</label>
+      <div class="p-inputgroup">
+        <span class="p-inputgroup-addon">
+          <i class="pi pi-map"></i>
+        </span>
+        <GMapAutocomplete
+          :placeholder="localOrgData.address?.formattedAddress ?? 'Enter an address'"
+          :options="{
+            fields: ['address_components', 'formatted_address', 'place_id', 'url'],
+          }"
+          class="p-inputtext p-component w-full"
+          data-cy="input-address"
+          @place_changed="setAddress"
+        >
+        </GMapAutocomplete>
+      </div>
+    </div>
+    <div v-if="orgType === 'districts'" class="form-field">
+      <label :class="{ 'font-light uppercase text-sm': !editMode }">NCES ID</label>
+      <div v-if="!editMode" :class="{ 'text-xl': !editMode }">{{ serverOrgData.ncesId ?? 'None' }}</div>
+      <PvInputText v-else v-model="localOrgData.ncesId" />
+    </div>
+    <div class="form-field">
+      <label :class="{ 'font-light uppercase text-sm': !editMode }">Tags</label>
+      <PvChips v-model="localOrgData.tags" />
+    </div>
+    <div>
+      <PvCheckbox v-model="localOrgData.testData" binary />
+      <label class="ml-1">Test Data</label>
+    </div>
+    <div>
+      <PvCheckbox v-model="localOrgData.demoData" binary />
+      <label class="ml-1">Demo Data</label>
+    </div>
+  </div>
+</template>
+<script setup>
+import { ref, defineProps, onMounted, watch } from 'vue';
+import { useAuthStore } from '@/store/auth';
+import { storeToRefs } from 'pinia';
+import { fetchDocById } from '@/helpers/query/utils';
+import { useQuery } from '@tanstack/vue-query';
+import _isEmpty from 'lodash/isEmpty';
+const props = defineProps({
+  orgType: { type: String, required: true },
+  orgId: { type: String, required: true },
+  editMode: { type: Boolean, default: true },
+});
+
+// +------------+
+// | Initialize |
+// +------------+
+const initialized = ref(false);
+const authStore = useAuthStore();
+const { roarfirekit } = storeToRefs(authStore);
+
+const emit = defineEmits(['modalClosed', 'update:orgData']);
+
+// +----------------------------+
+// | Query for existing orgData |
+// +----------------------------+
+const { data: serverOrgData } = useQuery({
+  queryKey: ['org', props.orgType, props.orgId],
+  queryFn: () => fetchDocById(props.orgType, props.orgId),
+  keepPreviousData: true,
+  enabled: initialized,
+  staleTime: 5 * 60 * 1000, // 5 minutes
+});
```

Comment:
What do you think of moving this query out of the component and encapsulate the query logic in a separate file as a reusable composable? This is usually the preferred approach when using something like Tanstack Query as it lets us re-use the queries, write some unit tests for it and overall make the code more maintainable. 

For example, something along the lines of: 

```js
// src/queries/useOrganizationQuery.vue
import { fetchDocById } from '@/helpers/query/utils';
import { useQuery } from '@tanstack/vue-query';
import { ORGANIZATION_QUERY_KEY } from 'constants/QueryKeys'

const useOrganizationQuery = (orgType, orgId, queryOptions = undefined) => {
  return useQuery({
    queryKey: [ORGANIZATION_QUERY_KEY, orgType, orgId],
    queryFn: () => fetchDocById(orgType, orgId),
    keepPreviousData: true,
    ...queryOptions
  });
}

export default useOrganizationQuery
```

```js
// src/constants/queryKeys.js
export const ORGANIZATION_QUERY_KEY = 'org'
```

Then, in any component where we want to use this query we can: 

```vue
<script setup>
import useOrganizationQuery from '@/queries/useOrganizationQuery'

// […]

const { data: orgData, isLoading: loadingOrgData } =
    useOrganizationQuery(
      orgType,
      orgId,
      {
        enabled: initialized
      }
    )

// […]
</script>
```

Additionally, the `staleTime` should ideally be configured globally as it's rarely changed on a case-by-case basis. Configuring a default globally in `main.js` is a good practice and then just override if it's necessary for a single query. 

Was planing on writing a ticket for this but noticed it in this PR, so figured I'd also raise this here.

What do you think?

Reply:
I really like this suggestion and I've added it as a task to the next sprint (assigned to @maximilianoertel). I think we should definitely do this but I don't think it should be a blocker for this current PR.

Reply:
Agree that this isn't something that shouldn't hold up the PR, that's why I left these comments as "comments" not "change requests" – apologies, should have made that clearer in the comments themselves. But thank you, happy to work on this on the next sprint!